src/hooks/useKeyFactory.js

```diff
@@ -0,0 +1,39 @@
+import { useMemo } from 'preact/hooks';
+
+const KEY_LENGTH = 6;
+
+/**
+ * Create a persistent key factory for plain objects without id.
+ *
+ * @example
+ * ```jsx
+ * function List({ objects }) {
+ *   const getKey = useKeyFactory();
+ *   return (<ol>{
+ *     objects.map(obj => {
+ *       const key = getKey(obj);
+ *       return <li key={key}>obj.name</li>
+ *     })
+ *   }</ol>);
+ * }
+ * ```
+ *
+ * @param {any[]} dependencies
+ * @returns {(element: object) => string}
+ */
+export function useKeyFactory(dependencies = []) {
```

Comment:
Not sure but just want to double-check: is there a need for a cleanup handler for this custom hook? Probably not because list entries are created from scratch once the selected element changes.

Reply:
No, because when the callback (`getKey`) is garbage collected, so is the map. It will be recreated from scratch once we render a new component.